src/util/cleaners.ts

```diff
@@ -26,3 +26,77 @@ export const asMaybeContractLocation = asMaybe(
     contractAddress: asString
   })
 )
+
+export const asObjectIn = <K extends string | number | symbol, T>(asKey: Cleaner<K>, asT: Cleaner<T>): Cleaner<{ [k in K]: T }> => {
+  return function asObject(raw) {
+    if (typeof raw !== 'object' || raw == null) {
+      throw new TypeError('Expected an object, got ' + showValue(raw))
+    }
+
+    const out: any = {}
+    const keys = Object.keys(raw)
+    for (let i = 0; i < keys.length; ++i) {
+      const key = keys[i]
+      if (key === '__proto__') continue
+      try {
+        out[key] = asT(raw[key])
+      } catch (error) {
+        throw locateError(error, '[' + JSON.stringify(key) + ']', 0)
+      }
+    }
+    return out
+  }
+}
+
+export const asObjectInOrBoolean = <K extends string | number | symbol, T>(asKey: Cleaner<K>, asT: Cleaner<T>) => asEither(asObjectIn(asKey, asT), asBoolean)
```

Comment:
Really you want `asObjectInOrTrue` since you can't create a `false` entry using the API anyway